imap_processing/ultra/l1b/ultra_l1b_extended.py

```diff
@@ -111,6 +123,115 @@ def get_front_y_position(start_type: ndarray, yb: ndarray) -> tuple[ndarray, nda
     return np.array(d), np.array(yf)
 
 
+def get_ph_tof_and_back_positions(
+    de_dataset: xarray.Dataset, xf: np.ndarray, sensor: str
+) -> tuple[np.ndarray, np.ndarray, np.ndarray, np.ndarray]:
+    """
+    Calculate back xb, yb position and tof.
+
+    An incoming particle may trigger pulses from one of the stop anodes.
+    If so, four pulses are produced, one each from the north, south,
+    east, and west sides.
+
+    The Time Of Flight (tof) and the position of the particle at the
+    back of the sensor are measured using the timing of the pulses.
+    Further description is available on pages 32-33 of
+    IMAP-Ultra Flight Software Specification document
+    (7523-9009_Rev_-.pdf).
+
+    Parameters
+    ----------
+    de_dataset : xarray.Dataset
+        Data in xarray format.
+    xf : np.array
```

Comment:
What is "xf" here. Is it a fixed value of the front position? Is it a part of the original de_dataset and you could index into that? How do we know it is the same shape as de_dataset (or maybe it isn't).